x/gamm/pool-models/stableswap/amm.go

```diff
@@ -76,6 +76,8 @@ func cfmmConstantMulti(xReserve, yReserve, u, v osmomath.BigDec) osmomath.BigDec
 // xy(x^2 + y^2) = (x - a)(y + b)((x - a)^2 + (y + b)^2)
 // and the following expression for `a` in multi-asset pools:
 // xyz(x^2 + y^2 + w) = (x - a)(y + b)z((x - a)^2 + (y + b)^2 + w)
+// TODO: explain where z comes from, it seems that we only use sum squares
+// in solveCFMMBinarySearchMulti (w).
```

Comment:
I thought that we are solving `h(x,y,w)` w/o `z`:

Ref:
https://github.com/osmosis-labs/osmosis/blob/593f0d2c101f2dc970255366333291a6c6a66203/x/gamm/pool-models/stableswap/README.md?plain=1#L129-L130

Reply:
Ah yeah that z is the doc is an accident, we need to delete it. Its not used within the function